src/topology.jl

```diff
@@ -104,3 +192,8 @@ function Cart_sub(comm::Comm, remain_dims)
     end
     comm_sub
 end
+
+function Cart_sub(comm::Comm, remain_dims)
```

Comment:
This is a bit of an inconsistent interface (and won't work on 32-bit machines where `Int == Cint`). One option would be to make it a keyword argument, i.e. `function Cart_sub(comm::Comm; keepdims=keepdims)` 

(it probably shouldn't be just `dims`, since this has basically the opposite behaviour in e.g. `mapslices` or `reduce`).

Reply:
Oh, i misunderstood what his was, nevermind.